File: tests/testuserfiles/test2.cpp
#include <iostream>

using namespace std;

void palindrome(const char *s) {
    if (s[0]) {
        cout << s[0];
        if (s[1]) {
            palindrome(s+1);
            cout << s[0];
        }
    }
}

int main() {
    const char *s = "foobar";
    palindrome(s);
    cout << endl;
}

File: tests/testuserfiles/test2.java
import java.util.*;
 
class Palindrome {
    public static void main(String args[]) {
        String original, reverse = "";
        
        original = "abcba";

        int length = original.length();

        for (int i = length - 1; i >= 0; i--)
        reverse = reverse + original.charAt(i);

        if (original.equals(reverse))
            System.out.println("The string is a palindrome.");
        else
            System.out.println("The string isn't a palindrome.");
    }
}